satpy/tests/test_file_handlers.py

```diff
@@ -21,7 +21,15 @@
 from unittest import mock
 import numpy as np
 
-from satpy.readers.file_handlers import BaseFileHandler
+from satpy.readers.file_handlers import BaseFileHandler, open_dataset
+
+
+def test_open_dataset():
+    """Test xr.open_dataset wrapper."""
+    fn = "path/to/file.nc"
```

Comment:
Should this be a MagicMock? that would allow to check if `open` was called on it.

Reply:
I had this mocked at first but then I thought that this probably was covered by the test of the `open_file_or_filename` function. I admit I didn't check but I have seen now that the function is not covered by a test. Should I mock it and test it here or should that be a separate test?

Reply:
well, what we want test here is that the dataset is opened whether it's a filename or a fsfile object, right? so I think both scenarios should be tested :)

Reply:
Yes the dataset should be opened regardless. But what I meant was; the logic for this is covered in `open_file_or_filename` and not here. That is why I assumed it should not necessarily be tested here. Anyway probably that is nitpicking as long as it is covered. I will add it here then.